tests/src/test/scala/cats/tests/FunctionSuite.scala

```diff
@@ -46,6 +46,22 @@ class FunctionSuite extends CatsSuite {
   // TODO: make an binary compatible way to do this
   // checkAll("Function1[Int => *]", DeferTests[Function1[Int, *]].defer[Int])
 
+  test("Defer[Function1[Int, *]].fix computing sum") {
+    val sum2 = Defer[Function1[Int, *]].fix[Int] {
+      rec =>
+        { n: Int =>
+          if (n <= 0) 0 else n * n + rec(n - 1)
+        }
+    }
+
+    forAll { n0: Int =>
+      val n = n0 & 0x3FF // don't let it get too big
```

Comment:
This is nitpicky but could you just write this out for clarity? I can't remember off the top of my head what this does for negative `n0` for example.

Reply:
It's taking the 18 least significant bits. Perhaps a slightly clearer way to write it for those who aren't doing bit manipulation on a regular basis would be `math.abs(n0 % 262144)`.

Reply:
I've changed it to be more direct `Gen.choose(0, 1000)` which I think is nicer anyway. Note `0x3ff = 1024`, it is actually 8 + 2 bits.